extensions/security/deployment/src/test/java/io/quarkus/security/test/permissionsallowed/checker/NoArgsPermissionCheckerValidationFailureTest.java

```diff
@@ -0,0 +1,48 @@
+package io.quarkus.security.test.permissionsallowed.checker;
+
+import jakarta.inject.Singleton;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.security.PermissionChecker;
+import io.quarkus.security.PermissionsAllowed;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class NoArgsPermissionCheckerValidationFailureTest {
```

Comment:
Very comprehensive test coverage, this is great.
Do you have a test which verifies that a checker which does not accept either `SecurityIdentity` or one of the JAX-RS method parameters is ignored ?

Reply:
> Do you have a test which verifies that a checker which does not accept either SecurityIdentity or one of the JAX-RS method parameters is ignored ?

It's this one (though "not ignored" but leads to build failure). Maybe I am missing something? If you mean a different scenario, please give me a hint.

TBH this test you commented on only exists because you explicitly requested it, personally I would allow no args because you can also inject `SecurityIdentity`, so if someone has bean with 5 checkers and they all use `SecurityIdentity` and he prefers to use `@Inject` instead of method argument, I would not prevent it. But this is such a small detail I suppose it is not worth any discussion at all. Let's keep it.

Reply:
Looks like it is this one, https://github.com/quarkusio/quarkus/pull/43846/files#diff-308cccee37a7dec2fc161afdbbfbbc2b922a91de72d6d1312faabf468467d550 ?

Reply:
ah, ok, you meant something different; you wanted to validate that all the unknown params are reported, yes, it is always reported as when we cannot find match on the side of Permission constructor, build fails.

Reply:
@michalvavrik See, we need some kind of evidence that the authenticated request can be authorized in that permission checker method. Otherwise users may end up doing some kind of self-authorization; there is no guarantee what that checker method will do it all correctly anyway but at least Quarkus will know the request specific info is available for that method.

Reply:
@michalvavrik What is going to happen here:
```
@PermissionAllowed("id")
void someJaxrsMethod(String id);

@PermissionChecker("id")
boolean checkId(long id) {}
```

?